packages/react/src/ReactElementValidator.js

```diff
@@ -214,14 +214,24 @@ function validateChildKeys(node, parentType) {
  *
  * @param {ReactElement} element
  */
+var propTypesMisspellWarningShown;
 function validatePropTypes(element) {
   var componentClass = element.type;
   if (typeof componentClass !== 'function') {
     return;
   }
   var name = componentClass.displayName || componentClass.name;
   var propTypes = componentClass.propTypes;
-
+  if (
+    componentClass.PropTypes !== undefined &&
+    !propTypesMisspellWarningShown
+  ) {
+    propTypesMisspellWarningShown = true;
+    warning(
+      false,
+      'Static propTypes method should be accessed by `.propTypes = { }` instead of `.PropTypes = { }`',
```

Comment:
Let's reword to say:

>Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?

And pass the component name `name  || 'Unknown'` as an argument.

Reply:
changes done. Thank you @gaearon for being so patient with me.